lib/rubocop/ast/node.rb

```diff
@@ -405,7 +405,9 @@ def immutable_literal?
               receiver.send(recursive_kind) &&
               arguments.all?(&recursive_kind)
           when :begin, :pair, *OPERATOR_KEYWORDS, *COMPOSITE_LITERALS
-            children.all?(&recursive_kind)
+            children.all? do |child|
+              child.nil? ? true : child.public_send(recursive_kind)
+            end
```

Comment:
`children.compact.all?(&recursive_kind)` should read simpler

Reply:
Awesome! I updated it.